sdk/compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -832,29 +835,46 @@ writeDalfFile dalfFile mod = do
                 Proto.toLazyByteString $
                     encodeSinglePackageModule lfVersion mod
 
+-- this is a lot of qualification, which to import?
+convertUnitId :: Map.Map GHC.UnitId LF.DalfPackage -> GHC.InstalledUnitId -> LF.PackageId
+convertUnitId pkgMap id =
+  --TODO: should we handle the case where lookup fails?
```

Comment:
In Dar.hs we return a Maybe and then use mapMaybe to drop anything that fails lookup, I'd be inclined to keep that behaviour. You're right that we'd expect this lookup to succeed in all cases.